src/commcare_cloud/ansible/roles/ecryptfs/tasks/main.yml

```diff
@@ -88,6 +88,25 @@
   tags:
     - ecryptfs-blobdb
 
+- name: Create {{encrypted_tmp}} to use for as tmp directory.
+  become: yes
+  file:
+    path: "{{ encrypted_tmp }}"
+    mode: "u=rwx,g=rwx,o=rx"
+    state: directory
+  tags:
+    - mount-ecryptfs
+    - after-reboot
+    - tmp-encrypt
+
+- name: Mount tmp drive
+  become: yes
+  shell: "mount -t ecryptfs -o key=passphrase:passphrase_passwd={{ ECRYPTFS_PASSWORD }},user,noauto,ecryptfs_cipher=aes,ecryptfs_key_bytes=32,ecryptfs_unlink_sigs,ecryptfs_enable_filename_crypto=y,ecryptfs_fnek_sig={{ password_hash.stdout }},verbosity=0 {{ encrypted_tmp }}/ {{ encrypted_tmp }}/"
+  when: (password_hash.stdout is defined) and ( encrypted_tmp not in mtab_contents.stdout )
```

Comment:
This just failed for me on hqdb1-staging when the directory was already mounted. (so did  "Mount data drive" which I think is where you got htis `when` from.)

Is there any reason we shouldn't be using hte 'mount' module?

Reply:
I think we don't use it because we don't want the password in `/etc/fstab` though I'm not even sure why we have it specified in the mount command if we're adding it to the kernel keyring. We also add the value of `ecryptfs_fnek_sig` to the mount options. I'm not sure if that value is a constant or different each time (I suspect it's different)